### ISSUE TITLE
Some operations/fragments validations are being skipped by `graphql-toolkit`

### ISSUE THREAD

**Issue by merisbahti** [dotansimha/graphql-code-generator#2452]
**Describe the bug**
The query (`Query`) gets the `__typename` `'Query' & 'Pizzeria'`, which is impossible.

**To Reproduce**
https://codesandbox.io/s/magical-dirac-nzc4c

1. My GraphQL schema:

```graphql
type OtherStuff {
  foo: String
}

type Pizzeria {
  id: Int,
  name: String,
  location: String,
}

type Query {
  otherStuff: OtherStuff
}
```
fragment:
```graphql
fragment pizzeriaFragment on Pizzeria {
  name
}
```

2. My GraphQL operations:

```graphql
query searchPage {
  otherStuff {
    foo
  }
  ...pizzeriaFragment 
}
```

3. My `codegen.yml` config file:

```yml
schema: src/schemas/*.graphql
generates:
  ./src/types.ts:
    documents:
      - src/queries/*.graphql
    plugins:
      - typescript
      - typescript-operations
    config:
      enumsAsTypes: true
      skipTypename: false
      preResolveTypes: true
```

**Expected behavior**
The type of SearchPageQuery should _not_ be `'Query' & 'Pizzeria'`.
**Environment:**

Sandbox

**Additional context**

**Comment by n1ru4l** [comment]
Does GraphQL allow spreading Fragments on types that are not covered by that fragment?

**Comment by merisbahti** [comment]
Sorry I'm not sure what you mean, could you provide a forked sandbox?

**Comment by n1ru4l** [comment]
The query is not valid: https://codesandbox.io/s/fervent-brahmagupta-60k48?fontsize=14

This should throw an error during code generation.

**Comment by dotansimha** [comment]
@ardatan can you please take a look? it seems like the codegen doesn't validate operations in this case

**Comment by dotansimha** [comment]
It seems like some rules with are ignoring in `graphql-toolkit` also causes it to skip the invalid type rule. I'll take a look and try to make the codegen throw this validation error.

**Comment by dotansimha** [comment]
Fixed in `graphql-toolkit`

**Comment by dotansimha** [comment]
Fixed in 1.7.0
